pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/rule/security/VfUnescapeElRule.java

```diff
@@ -0,0 +1,323 @@
+                    final List<ASTElExpression> elsInVal = attr.findDescendantsOfType(ASTElExpression.class);
+                    for (ASTElExpression el : elsInVal) {
+                        if (startsWithSlashLiteral(el)) {
+                            continue;
+                        }
+
+                        if (doesElContainAnyUnescapedIdentifiers(el, ESCAPING.URLENCODE)) {
+                            isEL = true;
+                            toReport.add(el);
+                        }
+                    }
+                }
+
+            }
+        }
+
+        if (isEL) {
+            for (ASTElExpression expr : toReport) {
+                addViolation(data, expr);
+            }
+        }
+
+    }
+
+    private boolean startsWithSlashLiteral(final ASTElExpression elExpression) {
+        final ASTExpression expression = elExpression.getFirstChildOfType(ASTExpression.class);
+        if (expression != null) {
+            final ASTLiteral literal = expression.getFirstChildOfType(ASTLiteral.class);
+            if (literal != null && literal.jjtGetChildIndex() == 0) {
+                if (literal.getImage().startsWith("'/") || literal.getImage().startsWith("\"/") || literal.getImage().toLowerCase().startsWith("'http") || literal.getImage().toLowerCase().startsWith("\"http") ) {
+                    return true;
+                }
+            }
+
+        }
+
+        return false;
+    }
+
+    private void checkTagsThatSupportEscaping(ASTElement node, Object data) {
+        final List<ASTAttribute> attributes = node.findChildrenOfType(ASTAttribute.class);
+        final Set<ASTElExpression> toReport = new HashSet<>();
+        boolean isUnescaped = false;
+        boolean isEL = false;
+        boolean hasPlaceholders = false;
+
+        for (ASTAttribute attr : attributes) {
+            String name = attr.getName().toLowerCase();
+            switch (name) {
+            case ESCAPE:
+            case ITEM_ESCAPED:
+                final ASTText text = attr.getFirstDescendantOfType(ASTText.class);
+                if (text != null) {
+                    if (text.getImage().equalsIgnoreCase(FALSE)) {
+                        isUnescaped = true;
+                    }
+                }
+                break;
+            case VALUE:
+            case ITEM_VALUE:
+
+                final List<ASTElExpression> elsInVal = attr.findDescendantsOfType(ASTElExpression.class);
+                for (ASTElExpression el : elsInVal) {
+                    if (doesElContainAnyUnescapedIdentifiers(el, ESCAPING.HTMLENCODE)) {
+                        isEL = true;
+                        toReport.add(el);
+                    }
+                }
+
+                final ASTText textValue = attr.getFirstDescendantOfType(ASTText.class);
+                if (textValue != null) {
+                    if (Pattern.compile("\\{(\\w|,|\\.|'|:|\\s)*\\}").matcher(textValue.getImage()).matches()) {
```

Comment:
consider moving the compiled pattern into a static final field. Regex compilation is a slow process to be done on the fly, specially for constant patterns.